src/pool/mod.rs

```diff
@@ -93,50 +68,43 @@ where
             }
         });
     }
 
-    /// Processes all current pending transactions by retrying them
-    /// and pruning them if necessary.
+    /// Processes all current pending transactions by pruning them if necessary.
+    ///
+    /// It also resets the `OpenZeppelin` account fleet.
     async fn process_pending_transactions(&self) -> Result<()> {
-        let pending_transactions = self.pending_transactions().await?;
-        for transaction in pending_transactions {
-            if self.should_retry(&transaction).await? {
-                self.retry_transaction(transaction).await?;
-            } else {
+        for transaction in self.pending_transactions().await? {
+            if self.should_prune(&transaction).await? {
                 self.prune_transaction(transaction.hash).await?;
             }
         }
         Ok(())
     }
 
-    /// Retries a transaction and prunes it if the conversion to a primitive transaction fails.
-    async fn retry_transaction(&self, transaction: StoredPendingTransaction) -> Result<()> {
-        let hash = transaction.hash;
-        tracing::info!(%hash, retries = transaction.retries + 1, "retrying");
-
-        // Generate primitive transaction, handle error if any
-        let transaction = match TransactionSignedEcRecovered::try_from(transaction.tx.clone()) {
-            Ok(transaction) => transaction,
-            Err(error) => {
-                self.prune_transaction(hash).await?;
-                return Err(error.into());
-            }
-        };
-
-        let _hash = self.eth_provider.send_raw_transaction(transaction.into_signed().envelope_encoded()).await?;
-        #[cfg(test)]
-        self.retried.lock().await.push(_hash);
-        Ok(())
-    }
-
     /// Prunes a transaction from the database given its hash.
+    ///
+    /// It also updates the associated `OpenZeppelin` account.
     async fn prune_transaction(&self, hash: B256) -> Result<()> {
         tracing::info!(%hash, "pruning");
+        // Construct a filter to delete the transaction from the database.
         let filter = EthDatabaseFilterBuilder::<filter::Transaction>::default().with_tx_hash(&hash).build();
+        // Delete the transaction from the database.
         self.database.delete_one::<StoredPendingTransaction>(filter).await?;
+
+        // Construct a filter to update the OZ account in the database.
+        let account_filter = EthDatabaseFilterBuilder::<filter::OzAccount>::default().with_tx_hash(&hash).build();
+        // Get the OZ account from the database.
+        let oz_account =
+            self.database.get_one::<StoredOzAccount>(account_filter.clone(), None).await?.unwrap_or_default();
+        // Remove the current transaction hash from the OZ account.
+        //
+        // The OZ account is now free to accept a new transaction.
+        self.database.update_one(StoredOzAccount { current_tx_hash: None, ..oz_account }, account_filter, true).await?;
+
```

Comment:
I'm not sure what is going on here

Reply:
In fact, what I did to try to reuse as much as possible what we had without breaking anything and being able to adapt, is the following approach:
- I replaced what was called the `RetryHandler` with a new name `PendingTxsHandler` (which I don't like too much the name and that we could change). But this service no longer takes care of retry because we no longer do this task but it processes pending transactions once they are executed.
- Indeed, for an OpenZeppelin account to be ready to receive a new transaction, it is necessary:
    - That it has a sufficient balance (> 1 ETH) and we check this at each turn in the accounts loop.
    - That it is free (i.e. that it has no more transactions in its pocket). For this I thought that the simplest approach was to store our 30 accounts in a collection `oz_accounts` of the database and to set a flag `current_tx_hash` which corresponds to `None` if it has no current transaction and to `Some(tx_hash)` if it has a current transaction with the given hash. This way we can easily see if the account is free or not and not overload it or create conflicting transactions on it.

And so in this context, I use our previous retry service to check, in an infinite loop, if the transactions are finalized (and therefore stored in the finalized database). If this is the case then I release the OpenZeppelin account. This is what we see here when I do `current_tx_hash: None` = release the account so that it is ready to receive new transactions.

We may eventually improve this system later, but at least it allows us to have a visualization of the transactions in progress in the database.

Reply:
Should be fixed via the new approach just fixed. I just did a scaffolding of the `start` method because it will probably be subject to changes when we integrate the OZ accounts and test it. So we shouldn't worry too much about its structure for the moment I think, we'll take care of it later in the follow ups PRs when we fill the process_transaction.